src/vue/components/nav/action-panel.vue

```diff
@@ -21,7 +21,7 @@
       />
         <progress-bar
             class="alpheios-action-panel__progress-bar"
-            v-if="$store.getters['app/lexicalRequestInProgress']"
+            v-if="!$store.state.app.morphDataReady"
```

Comment:
I think this will not work as expected. As in the current version, it will display a progress bar whenever morph data is not available. When we first open the page, the morph data is not there, and the progress bar will be shown.

To test it: open an embedded lib in Chrome's dev tools, switch to mobile emulation so that the compact version of a toolbar (a single button) be shown. Click on a toolbar button. It will open an action panel, and the progress bar will be running there.

Right now it relies on `lexicalRequestInProgress` getter. When request is started, we set `state.lexicalRequest.startTime`. `state.lexicalRequest.endTime` contains either the time when the last lexical request was completed or zero if an app is just initialized. So the function is like: `lexicalRequestInProgress (state) { return state.lexicalRequest.startTime > state.lexicalRequest.endTime }`.

I'm not fully familiar with the word list logic, but I think we should either:
1. Not change `state.lexicalRequest.startTime` when we select a word from the word list;
2. Set `state.lexicalRequest.endTime` to the current time when the word list selection is done. This will stop a progress indicator.
What do you think?

Reply:
For now we have three progress-bar - in panel, popup and actionPanel
In panel and popup we use this condition - `!$store.state.app.morphDataReady`
and it works well

But only for the actionPanel - we have another condition - that doesnt work well for selected word list items.
I have made it the same here - and tested in embed-lib the same way you have described - it works.

Why don't you think, that it shouldn't work here?
Why do you need here in actionPanel another condition?

When we get words from the wordList we don't do any lexiqal request (it the word is from the current session) - so there are no any startTime or endTime.
It has the following:
```
this.onHomonymReady(homonym)
this.updateDefinitions(homonym)
this.updateTranslations(homonym)
```

And I think that it is correct to check morphDataReady - because it shows, that we have got data from LexiqalRequest
Why do you think that checking time is better?

Reply:
Let me test and think about it for a while, I'm not very familiar with the word list code at the moment

Reply:
> I have made it the same here - and tested in embed-lib the same way you have described - it works.

Does it work for you in the following situation:
1. Open an embedded lib page in mobile device emulation mode
2. Reload the page.
3. Click on the toolbar's button.
For me it shows an action panel with the toolbar running:
![image](https://user-images.githubusercontent.com/18631055/57856417-0ae9be00-77fe-11e9-8dc3-850f1e31c483.png)

Reply:
In the popup, we have a condition for the progress bar as `!$store.state.app.morphDataReady && !noLanguage`, in the panel - `!$store.state.app.morphDataReady && Boolean(this.$store.state.app.currentLanguageName)` which is, I believe, pretty much the same.

But the main difference, however, is that we open popup or panel only when the lexical request is started. If we, after loading an embedded lib's page in the desktop mode and doing nothing else with it will manually set `display: block` for the popup, we will see that the progress bar running there. So the conditions in panel or popup does not prevent the progress bar from being displayed while in idle. We don't need it because panel and popup hides it by not displaying itself.

The action panel, on the opposite, is always visible. That's why we need different display condition for the progress bar there.

Reply:
When we click on the word list items, it calls `selectWordItem()` of a word list controller. I'm not 100% sure yet, but probably it somehow sets `state.lexicalRequest.startTime` to the current time which makes the progress bar to run (probably not by itself, but maybe by calling one of the UI controller's methods that do that). If this is true, then we need just to prevent it from touching  `state.lexicalRequest.startTime` and that should fix the issue.

I will continue an investigation.

Reply:
`uiController.onWordItemSelected` respondes to `selectWordItem()`. This calls `uiController.newLexicalRequest` which is where `state.lexicalRequest.startTime` gets touched.

Reply:
Then, if we do a "real" lexical query there, maybe we should call a `lexicalRequestFinished()` mutation in `uiController.onWordItemSelected()` after we're done? It will set the `lexicalRuquest.endTime` and fix the progress bar issue. What do you think?

Reply:
Oh, I understood know what you mean, I think that then we should sync conditions for all three progress bar - to check if lexicalQuery is working.

Reply:
I agree that display conditions should be the same for all three bars, if possible.

Reply:
Fixed